src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -102,16 +102,15 @@ public async Task<IActionResult> List(ListContentsViewModel model, PagerParamete
                 }
 
                 // Allows non creatable types to be created by another admin page.
-                if (model.Options.CanCreateSelectedContentType)
+                if (contentTypeDefinition.GetSettings<ContentTypeSettings>().Creatable || model.Options.CanCreateSelectedContentType)
                 {
                     model.Options.CreatableTypes = new List<SelectListItem>
                     {
                         new SelectListItem(new LocalizedString(contentTypeDefinition.DisplayName, contentTypeDefinition.DisplayName).Value, contentTypeDefinition.Name)
                     };
                 }
             }
-
-            if (model.Options.CreatableTypes == null)
+            else
```

Comment:
Then, if `CreatableTypes` is necessarily initialized when there is a `SelectedContentType` (see previous comment), here we could re-use `if (model.Options.CreatableTypes == null)`

This would still allow, if no `SelectedContentType`, to preset `CreatableTypes` e.g. with a reduced set of types.